src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -109,23 +124,103 @@ private static synchronized void closeResources()  {
       zoneIdToTable.clear();
       zoneIdToTable = null;
     }
-    if (fixedTransitions != null) {
-      fixedTransitions.close();
-      fixedTransitions = null;
+    if (transitions != null) {
+      transitions.close();
+      transitions = null;
     }
   }
 
-  public static ColumnVector fromTimestampToUtcTimestamp(ColumnVector input, ZoneId currentTimeZone) {
-    // TODO: Remove this check when all timezones are supported
-    // (See https://github.com/NVIDIA/spark-rapids/issues/6840)
-    if (!isSupportedTimeZone(currentTimeZone)) {
-      throw new IllegalArgumentException(String.format("Unsupported timezone: %s",
-          currentTimeZone.toString()));
+  private static long getScaleFactor(ColumnVector input){
+    DType inputType = input.getType();
+    if (inputType == DType.TIMESTAMP_SECONDS){
+      return 1;
+    } else if (inputType == DType.TIMESTAMP_MILLISECONDS){
+      return 1000;
+    } else if (inputType == DType.TIMESTAMP_MICROSECONDS){
+      return 1000*1000;
+    }
+    throw new UnsupportedOperationException("Unsupported data type: " + inputType);
+  }
+
+  public static boolean isSupportedTimeZone(String zoneId) {
+    try {
+      getZoneId(zoneId); // check that zoneId is a valid zone
+      return true;
+    } catch (ZoneRulesException e) {
+      return false;
+    }
+  }
+
+  // enforce that all timestamps, regardless of timezone, be less than the desired date
+  private static boolean shouldFallbackToCpu(ColumnVector input, ZoneId zoneId){
```

Comment:
This method is not named according to the logic. It actually returns **true** for inputs where the computation should run on GPU, and **false** for inputs that will be copied to host and the computation will run on the CPU.

I would suggest either renaming it to `canRunOnGpu` or reversing the logic to return true for CPU and false for GPU.